spot-client/src/common/app-state/remote-control-service/actions.js

```diff
@@ -79,3 +104,61 @@ export function setVideoMute(mute) {
                 requestTypes.VIDEO_MUTE, requestStates.ERROR)));
     };
 }
+
+/**
+ * Start wireless screensharing.
+ *
+ * @returns {Function}
+ */
+export function setWirelessScreensharing() {
+    return dispatch => {
+        dispatch(setRequestState(
+            requestTypes.SCREENSHARE,
+            requestStates.PENDING,
+            'proxy'
+        ));
+
+        return remoteControlService.setWirelessScreensharing(true, {
+            onClose: () => dispatch(setLocalWirelessScreensharing(false))
+        })
+            .then(() => {
+                dispatch(setLocalWirelessScreensharing(true));
+                dispatch(setSpotTVState({ screensharingType: 'proxy' }));
+                dispatch(setRequestState(
+                    requestTypes.SCREENSHARE,
+                    requestStates.DONE,
+                    'proxy'
+                ));
+            })
+            .catch(() => dispatch(setRequestState(
+                requestTypes.SCREENSHARE, requestStates.ERROR)));
+    };
+}
+
+/**
+ * Stops any screenshare in progress on the Spot-TV.
+ *
+ * @returns {Function}
+ */
+export function stopScreenshare() {
```

Comment:
what happens when `stopScreenshare()` is called while `remoteControlService.setWirelessScreensharing` called above is still working ? Is that even possible ? Will the start promise get rejected ? Im wondering what will happen to the request state

Reply:
is that a bug ?

Reply:
Yeah

Reply:
Hid the UI while the work is still being done.